docs/source/app_developers_guide/about_events.rst

```diff
@@ -0,0 +1,123 @@
+*********************
+About Sawtooth Events
+*********************
+
+Sawtooth events occur "on block boundaries" - that is, when blocks are
+committed - and are not persisted in state. Each transaction family can define
+the events that are appropriate for its business logic.
+
+Events are represented with the following protobuf message:
+
+.. code-block:: python
+
+   message Event {
+     // Used to subscribe to events and servers as a hint for how to deserialize
+     // event_data and what pairs to expect in attributes.
+     string event_type = 1;
+
+     // Transparent data defined by the event_type.
+     message Attribute {
+       string key = 1;
+       string value = 2;
+     }
+     repeated Attribute attributes = 2;
+
+     // Opaque data defined by the event_type.
+     bytes  data = 3;
+   }
+
+The ``event_type`` field (the name of the event) is used to determine how opaque
+(application-specific) data has been serialized and what transparent event
+attributes to expect.
+
+For more information, see
+:doc:`/architecture/events_and_transactions_receipts`.
+
+
+Event Namespace
+===============
+
+By convention, event names use the transaction family as a prefix, such as
+``xo/create`` for a create event from the XO transaction family.
+
+Core Sawtooth events are prefixed with ``sawtooth/``. The core events are:
+
+  * ``sawtooth/block-commit``
+
+  * ``sawtooth/state-delta``
+
+sawtooth/block-commit
+---------------------
+
+A ``sawtooth/block-commit`` event occurs when a block is committed. This event
+contains information about the block, such as the block ID, block number, state
+root hash, and previous block ID. It has the following structure:
+
+.. code-block:: protobuf
+
+   Event {
+     event_type = "sawtooth/block-commit",
+     attributes = [
+       Attribute { key = "block_id", value = "abc...123" },
+       Attribute { key = "block_num", value = "523" },
+       Attribute { key = "state_root_hash", value = "def...456" },
+       Attribute { key = "previous_block_id", value = "acf...146" },
+     ],
+   }
+
+sawtooth/state-delta
+--------------------
+
+A ``sawtooth/state-delta`` occurs when a block is committed. This event
+contains a single change in state at a given address. It has the following
```

Comment:
This may have changed, but I remember this returned all state changes that occurred in this block, not a single change.